Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1351,7 +1351,23 @@ public boolean cantBeAttachedBy(MageObject attachment, Ability source, Game game
                 return !ability.getDoesntRemoveControlled() || isControlledBy(game.getControllerId(attachment.getId()));
             }
         }
-        return game.getContinuousEffects().preventedByRuleModification(new StayAttachedEvent(this.getId(), attachment.getId(), source), null, game, silentMode);
+
+        boolean canAttach = true;
+        Permanent attachmentPermanent = game.getPermanent(attachment.getId());
+        if (attachment.hasSubtype(SubType.AURA, game)
+                && attachmentPermanent != null
+                && attachmentPermanent.getSpellAbility() != null
+                && !attachmentPermanent.getSpellAbility().getTargets().isEmpty()) {
+            Filter filter = attachmentPermanent.getSpellAbility().getTargets().get(0).getFilter();
+            if (filter instanceof FilterPermanent) {
+                canAttach = ((FilterPermanent) filter).match(this, attachmentPermanent.getControllerId(), source, game);
```

Comment:
Code with direct filter  looks weird (ai uses same workaround to find possible targets, but it’s a not good style). Only one target, only one filter type support. Why not to use methods target methods like canTarget, stillValid, etc?

Reply:
See code above as example: ability.canTarget(attachment, game)

Reply:
Is 4a67c66 what you had in mind? @JayDi85

Reply:
Yes. And no need to check filter anymore:
![IMG_0201](https://github.com/magefree/mage/assets/8344157/8452045b-8470-48a9-8c0b-5383566b918d)

Reply:
Done in 641d0b6